src/scripts/autocomplete.js

```diff
@@ -0,0 +1,499 @@
+    }
+  }
+
+  // Select project
+  if (!!ids) {
+    this.el.querySelector("li[data-pid='" + ids + "']").classList.add("selected-" + this.type);
+  }
+};
+
+ProjectAutoComplete.prototype.selectTask = function (elem, silent) {
+  // Set selected task
+  var currentSelected = this.el.querySelector(".selected-task");
+
+  if (!!currentSelected) {
+    currentSelected.classList.remove("selected-task");
+  }
+  elem.classList.add("selected-task");
+
+  // Set selected project
+  this.selectProject(elem.parentNode.parentNode, silent, true);
+};
+
+ProjectAutoComplete.prototype.selectProject = function (elem, silent, removeTask) {
+  if (elem.classList.contains("item-name")) {
+    elem = elem.parentNode;
+  }
+  if (!elem.classList.contains(this.type + "-row")) {
+    if (elem.classList.contains("task-count")) {
+      this.toggleTaskList(elem);
+    }
+    if (elem.classList.contains("task-item")) {
+      this.selectTask(elem);
+    }
+    return;
+  }
+
+  if (!removeTask && this.el.querySelector(".selected-task")) {
+    this.el.querySelector(".selected-task").classList.remove("selected-task");
+  }
+
+  var currentSelected = this.el.querySelector(".selected-" + this.type),
+    val = elem.getAttribute("data-pid");
+
+  if (!!currentSelected) {
+    currentSelected.classList.remove("selected-" + this.type);
+  }
+  elem.classList.add("selected-" + this.type);
+
+  // Update placeholder
+  this.placeholderDiv.textContent = this.placeholderDiv.title = this.generateLabel(this.getSelected(), val, this.type);
+  this.setProjectBullet(val);
+
+  if (!silent) {
+    // Close dropdown
+    this.closeDropdown();
+  }
+  return false;
+};
+
+AutoComplete.prototype.toggleTaskList = function (elem) {
+  var opened = this.el.querySelector(".tasklist-opened");
+  if (!!opened) {
+    opened.classList.remove("tasklist-opened");
+  }
+  if (opened !== elem.parentNode) {
+    elem.parentNode.classList.toggle("tasklist-opened");
+  }
+};
+
+ProjectAutoComplete.prototype.getSelected = function () {
+  var selected = this.el.querySelector(".selected-" + this.type),
+    task = this.el.querySelector(".selected-task"),
+    pid = (!!selected) ? parseInt(selected.getAttribute("data-pid"), 10) : 0,
+    tid = (!!task) ? parseInt(task.getAttribute("data-tid"), 10) : null,
+    name = (!!selected) ? selected.getAttribute("title") : "";
+
+  return {
+    el: selected,
+    pid: pid,
+    tid: tid,
+    name: name
+  };
+};
+
+ProjectAutoComplete.prototype.getSelectedProjectByPid = function (pid) {
+  var selected = this.el.querySelector("li[data-pid='" + pid + "']"),
+    name = (!!selected) ? selected.textContent : "";
+
+  return {
+    el: selected,
+    pid: pid,
+    name: name
+  };
+};
+
+ProjectAutoComplete.prototype.setProjectBullet = function (pid, tid, el) {
+  var project,
+    elem = el || this.placeholderItem.querySelector(".project-bullet"),
+    result,
+    task;
+
+  if (!!pid || pid === "0") {
```

Comment:
In the server request return if project is not set it is undefined. This is why `!!pid` works. Also added `!!tid` check.